### PR TITLE
Malfunctioning AI, TGUI menu and action buttons

### PR DESCRIPTION
**Pull request by Tsar-Salat**
Ports:
- https://github.com/tgstation/tgstation/pull/49532
- https://github.com/tgstation/tgstation/pull/49625

## About The Pull Request

As the title says. This PR does two things:

- Removes the 'Malfunctions' tab in the statuspanel, and moves it to an action button.
image

- Adds a TGUI menu for picking malf modules
image

Also cleans up miscellaneous code about abilities not clearing from AI's correctly.

## Why It's Good For The Game
Dear god that UI was awful

## Testing Photographs and Procedure

The menu in use.

https://discord.com/channels/427337870844362753/943311132007997490/1126747463001321492

## Changelog
:cl: RKz, Arkatos1, 
add: added new TGUI menu for Malfunctioning AI abilties!
add: added new action buttons for Malfunctioning AI
remove: removed management of malf AI abilities through the godawful statuspanel tab
rscadd: new related icon for action button

/:cl:

### CODE REVIEW

**Comment by Tsar-Salat** [comment]
bro is NOT pretty!

**Comment by Tsar-Salat** [comment]
pretty now

**Comment by Absolucy** [comment]
you should integrated this into the `AntagInfoMalfunction` UI - which I believe tg ended up doing, too.

**Comment by Tsar-Salat** [comment]
> you should integrated this into the `AntagInfoMalfunction` UI - which I believe tg ended up doing, too.

Yes. I looked into that. Specifically https://github.com/tgstation/tgstation/pull/62774 .

However as you said in the antag popup port #9205 , flavor was omitted.

Theres nothing really for me to build on. That menu would contain information about employer and background. Had flavor been ported in that pr, there would be reason to combine the menus as tg has done

I prefer to leave any future flavor in a separate pr as you have done.

File: code/modules/mob/living/silicon/ai/ai.dm [hunk 1]
```diff
@@ -205,8 +206,17 @@
...
 	malfhack = null
 	ShutOffDoomsdayDevice()
+
 	. = ..()
 
+/// Removes all malfunction-related abilities from the AI
+/mob/living/silicon/ai/proc/remove_malf_abilities()
+	QDEL_NULL(modules_action)
+	for(var/datum/AI_Module/AM in current_modules)
+		for(var/datum/action/A in actions)
```
Comment (Penwin0) [inline code/modules/mob/living/silicon/ai/ai.dm:228]: Can we use as () or as anything here? if so please do

File: code/modules/mob/living/silicon/ai/ai.dm [hunk 2]
```diff
@@ -891,49 +891,71 @@
...
 	to_chat(src, "You are also capable of hacking APCs, which grants you more points to spend on your Malfunction powers. The drawback is that a hacked APC will give you away if spotted by the crew. Hacking an APC takes 60 seconds.")
 	view_core() //A BYOND bug requires you to be viewing your core before your verbs update
-	add_verb(/mob/living/silicon/ai/proc/choose_modules)
 	malf_picker = new /datum/module_picker
-
+	modules_action = new(malf_picker)
+	modules_action.Grant(src)
 
 /mob/living/silicon/ai/reset_perspective(atom/A)
+	//SHOULD_CALL_PARENT(FALSE)
```
Comment (Penwin0) [inline code/modules/mob/living/silicon/ai/ai.dm:978]: Commented code

Comment (Tsar-Salat) [reply code/modules/mob/living/silicon/ai/ai.dm:978]: intended

Comment (Penwin0) [reply code/modules/mob/living/silicon/ai/ai.dm:978]: Why is it here then

**Comment by itsmeow** [comment]
I *hate* the way the TGUI malfunction theme looks.. if you wanted to try tweaking it that would be epic

File: code/modules/mob/living/silicon/ai/ai.dm (hunk 1 above)
Comment (itsmeow) [reply code/modules/mob/living/silicon/ai/ai.dm:228]: ~~It's not a heavy enough proc to warrant removing null safety tbh~~ lmao it literally checks istype right after. yea do that

**Comment by Tsar-Salat** [comment]
im taking a break, Semester begins soon and this is affecting my mental health

**Pull request closed without merging**